clinica/pipelines/engine.py

```diff
@@ -772,7 +776,21 @@ def copy2_add_ses(src, dst):
                 cprint(
                     msg=(
                         f"{len(cross_subj)} subjects in your BIDS folder did not respect the longitudinal organisation "
-                        "from BIDS specification. Clinica does not know how to handle cross sectional dataset, but it "
+                        "from BIDS specification."
+                    ),
+                    lvl="warning",
+                )
+                for subj in cross_subj:
```

Comment:
This is generating a warning message for each problematic subject which might be very very verbose if there are lots of them.
It might be more readable to build a bullet list of these subjects and append it to the warning message. WDYT?

Reply:
I agree that spawning dozens of warnings in case the dataset happens to have a lot of these errors would be too verbose. Besides, `print` statements are not cheap regardless of the language.

Now I wonder whether we should limit the upper bound to this list in case there is like 50 or a 100 issues in a large dataset for instance. Perhaps that's a separate concern though.

Reply:
yup it makes sense, I have done something, tell me what you think of it :)

Reply:
I'm not sure to understand why you are not building the full warning message string, and then pass it to `cprint`. Unless I'm missing something, this is still potentially giving the user tons of warnings. 

Maybe we could do something along those lines:

```python
msg = f"{len(cross_subj)} subjects in your BIDS....The subjects concerned are:\n\t- "  # this is fine since you know there's at least one
msg += "\n\t- ".join(cross_subj)
msg += f"\nClinica does not know...."
cprint(msg, lvl="warning")
```

Then, it should be straightforward to bound the list if needed as suggested by @ghisvail

Reply:
On principle, we should aim to decouple business logic from error handling logic. In practice, this means this sort of formatting code should belong to some domain exception class (let's say `ClinicaInconsistentDatasetError`) constructed with the relevant context (here a list of subjects) and responsible for constructing the appropriate error message.

Then the callee just needs to initialize the exception class with the collected data (the list of subjects not conforming) and the caller is responsible for catching the exception and forwarding the exception message to the appropriate logging channel. It makes also the error handling code significantly more testable, whereas the current code is not.

Clinica is filled with procedural code doing some sort of clumsy error reporting. Perhaps this PR is the right opportunity to experiment with more Pythonic error handling mechanisms, considering how localized it is.

Reply:
@ghisvail I agree this would be a good thing to do indeed.
I don't think this is too much work, so we should probably do it in this PR.

@MatthieuJoulot I think we need to create a `ClinicaInconsistentDatasetError` class in `clinica/utils/exceptions.py` and simply move all this formatting logic inside such that the engine would look like:

```python
if len(cross_subj) > 0:
    raise ClinicaInconsistentDatasetError(cross_subs)
```

Reply:
@NicolasGensollen @ghisvail this may be a naive comment, but won't raising an error kill the process ? which we wouldn't want to because the user may want to create an alternative dataset.

Reply:
> but won't raising an error kill the process ?

If you don't catch it, indeed.

That's why you'd need to surround the call to the function with a `try / except` block catching `ClinicaInconsistentDatasetError` and forwarding the error message to `cprint` with warning level.

Reply:
Okay ! Works in part, but since `check_not_cross_sectionnal` is a very long function with lots of stuff inside (bit of mess), currently it would still break stuff. I'm going to look for way to isolate the specific part we are interested in.

Reply:
Okay @ghisvail @NicolasGensollen I created a class and tried to handle the `try / except` in a smart way. WDYT ?